table/block_based/block_based_table_builder.h

```diff
@@ -101,6 +106,9 @@ class BlockBasedTableBuilder : public TableBuilder {
  private:
   bool ok() const { return status().ok(); }
 
+  template <typename T>
```

Comment:
The template feels an overkill to me. It is always Status, isn't it?

Reply:
This is for perfect forwarding for accepting both l-value Status and r-value Status as its argument. But you're right, we can make it simpler by assigning to global status only in non-OK cases.